src/mlpack/tests/wgan_test.cpp

```diff
@@ -0,0 +1,290 @@
+      1, 1, 1, 1, 7, 7);
+  generator.Add<BatchNorm<> >(6272);
+  generator.Add<ReLULayer<> >();
+  generator.Add<TransposedConvolution<> >(dNumKernels, 1, 15, 15, 1, 1, 1, 1,
+      14, 14);
+  generator.Add<TanHLayer<> >();
+
+  // Create WGAN
+  GaussianInitialization gaussian(0, 1);
+  Adam optimizer(stepSize, batchSize, 0.9, 0.999, eps, numIterations,
+      tolerance, shuffle);
+  std::function<double()> noiseFunction = [] () {
+      return math::RandNormal(0, 1);};
+  GAN<FFN<EarthMoverDistance<> >, GaussianInitialization,
+      std::function<double()>, WGAN> wgan(trainData, generator, discriminator,
+      gaussian, noiseFunction, noiseDim, batchSize, generatorUpdateStep,
+      discriminatorPreTrain, multiplier, clippingParameter);
+
+  Log::Info << "Training..." << std::endl;
+  wgan.Train(optimizer);
+
+  // Generate samples
+  Log::Info << "Sampling..." << std::endl;
+  arma::mat noise(noiseDim, batchSize);
+  size_t dim = std::sqrt(trainData.n_rows);
+  arma::mat generatedData(2 * dim, dim * numSamples);
+
+  for (size_t i = 0; i < numSamples; i++)
+  {
+    arma::mat samples;
+    noise.imbue( [&]() { return noiseFunction(); } );
+
+    generator.Forward(noise, samples);
+    samples.reshape(dim, dim);
+    samples = samples.t();
+
+    generatedData.submat(0, i * dim, dim - 1, i * dim + dim - 1) = samples;
+
+    samples = trainData.col(math::RandInt(0, trainData.n_cols));
+    samples.reshape(dim, dim);
+    samples = samples.t();
+
+    generatedData.submat(dim,
+        i * dim, 2 * dim - 1, i * dim + dim - 1) = samples;
+  }
+
+  Log::Info << "Output generated!" << std::endl;
+}
+
+/*
+ * Tests the gradient-penalized WGAN implementation on the MNIST dataset.
+ * It's not viable to train on bigger parameters due to time constraints.
+ * Please refer mlpack/models repository for the tutorial.
+ */
+BOOST_AUTO_TEST_CASE(WGANGPMNISTTest)
+{
+  size_t dNumKernels = 32;
+  size_t discriminatorPreTrain = 5;
+  size_t batchSize = 5;
+  size_t noiseDim = 100;
+  size_t generatorUpdateStep = 1;
+  size_t numSamples = 10;
+  double stepSize = 0.0003;
+  double eps = 1e-8;
+  size_t numEpoches = 1;
+  double tolerance = 1e-5;
+  int datasetMaxCols = 10;
+  bool shuffle = true;
+  double multiplier = 10;
+  double clippingParameter = 0.01;
+  double lambda = 10.0;
+
+  Log::Info << std::boolalpha
+      << " batchSize = " << batchSize << std::endl
+      << " generatorUpdateStep = " << generatorUpdateStep << std::endl
+      << " noiseDim = " << noiseDim << std::endl
+      << " numSamples = " << numSamples << std::endl
+      << " stepSize = " << stepSize << std::endl
+      << " numEpoches = " << numEpoches << std::endl
+      << " tolerance = " << tolerance << std::endl
+      << " shuffle = " << shuffle << std::endl;
+
+  arma::mat trainData;
+  trainData.load("mnist_first250_training_4s_and_9s.arm");
+  Log::Info << arma::size(trainData) << std::endl;
+
+  if (datasetMaxCols > 0)
```

Comment:
We could remove the check here to solve the static code analysis issue, same for the other tests.